### PR TITLE
CLI | Add threshold user input validation  (AST-40169)

### PR DESCRIPTION
**Pull request by AlvoBen**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

Add threshold user input validation

### References

https://checkmarx.atlassian.net/browse/AST-40169

### Testing

* Run tests localy
* Added unit tests

### Checklist

- [x] I have added documentation for new/changed functionality in this PR (if applicable).
- [x] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

### CODE REVIEW

File: internal/commands/scan.go [hunk 1]
```diff
@@ -2492,9 +2486,49 @@ func validateCreateScanFlags(cmd *cobra.Command) error {
...
+}
+
+func normalizeThresholds(threshold string) string {
+	return strings.ReplaceAll(strings.ReplaceAll(threshold, " ", ""), ",", ";")
+}
+
+func parseThresholdLimit(limit string) (engineName string, intLimit int, err error) {
+	parts := strings.Split(limit, "=")
+	engineName = strings.Replace(parts[0], commonParams.KicsType, commonParams.IacType, 1)
+	if len(parts) > 1 {
```
Comment (checkmarx-kobi-hagmi) [inline internal/commands/scan.go:2523]: You can reduce Cyclomatic Complexity if you invert the condition:
if len(parts) <= 1
{
return engineName, intLimit, err
}

and later the logic you currently have inside len(parts) > 1

File: internal/commands/scan.go [hunk 2]
```diff
@@ -2492,9 +2486,49 @@ func validateCreateScanFlags(cmd *cobra.Command) error {
...
+		}
+		if intLimit < 1 {
+			return errors.Errorf("%s: Threshold limit should be greater or equal to 1\n", engineName)
+		}
+	}
+	return nil
+}
+
+func normalizeThresholds(threshold string) string {
+	return strings.ReplaceAll(strings.ReplaceAll(threshold, " ", ""), ",", ";")
```
Comment (checkmarx-kobi-hagmi) [inline internal/commands/scan.go:2517]: Suggestion:
any chance to make it more readable?

File: internal/commands/scan.go [hunk 3]
```diff
@@ -2492,9 +2486,50 @@ func validateCreateScanFlags(cmd *cobra.Command) error {
...
+	return nil
+}
+
+func rearrangeUserInputThresholds(threshold string) string {
+	return strings.ReplaceAll(strings.ReplaceAll(threshold, " ", ""), ",", ";")
+}
+
+func parseThresholdLimit(limit string) (engineName string, intLimit int, err error) {
+	parts := strings.Split(limit, "=")
+	engineName = strings.Replace(parts[0], commonParams.KicsType, commonParams.IacType, 1)
```
Comment (OrShamirCM) [inline internal/commands/scan.go:2522]: Are we supporting only kicks and iac?

File: internal/commands/scan.go [hunk 4]
```diff
@@ -2492,9 +2486,50 @@ func validateCreateScanFlags(cmd *cobra.Command) error {
...
+
 	return nil
 }
 
+func validateThresholds(cmd *cobra.Command) error {
+	threshold, _ := cmd.Flags().GetString(commonParams.Threshold)
+	if threshold == "" {
+		return nil
+	}
+	threshold = rearrangeUserInputThresholds(threshold)
```
Comment (OrShamirCM) [inline internal/commands/scan.go:2502]: there is a change we are doing the rearrange twice? cause i see it called when starting a scan too

Comment (AlvoBen) [reply internal/commands/scan.go]: No, this line replace the word kicks with iac-security

File: internal/commands/scan.go (hunk 3 above)
Comment (AlvoBen) [reply internal/commands/scan.go:2514]: No, this line replace the word kicks with iac-security

**Review by OrShamirCM — APPROVED** [review]

**Pull request merged**